src/native/cipher.rs

```diff
@@ -261,3 +248,297 @@ pub fn decrypt_in<P: PoolParams>(eta:Num<P::Fr>, memo:&[u8], params:&P)->Vec<Opt
         vec![]
     }
 }
+
+/// get encrypted memo chunks with associated decryption keys (chunk: account or note)
+/// returns vector of tupple (index, chunk, key)
+/// indexes are zero-based and enumerated within current memo
+pub fn symcipher_decryption_keys<P: PoolParams>(eta:Num<P::Fr>, mut memo:&[u8], params:&P) -> Option<Vec<(u64, Vec<u8>, Vec<u8>)>> {
+    let num_size = constants::num_size_bits::<P::Fr>()/8;
+    let account_size = constants::account_size_bits::<P::Fr>()/8;
+    let note_size = constants::note_size_bits::<P::Fr>()/8;
+
+    let nozero_items_num = u32::deserialize(&mut memo).ok()? as usize;
+    if nozero_items_num == 0 {
+        return None;
+    }
+
+    let nozero_notes_num = nozero_items_num - 1;
+    let shared_secret_ciphertext_size = nozero_items_num * constants::U256_SIZE + constants::POLY_1305_TAG_SIZE;
```

Comment:
Why do we store Poly1305 AEAD Tag if it's not used in decryption? 
That could save 16 bits in memo

Reply:
> Why do we store Poly1305 AEAD Tag if it's not used in decryption?

This tag is used for authentication of the encrypted message. It performed under the hood of ChaChaPoly algorithm. [Here](https://github.com/zkBob/libzeropool-zkbob/blob/02f0b6ce4e737abfc99858ea01b4ddc86ec5ee90/src/native/cipher.rs#L66) we pass the associated data (plain) with encrypted data as well. It's the core of AEAD mode

> That could save 16 bits in memo

16 bytes :)